Fix a couple of deployment bugs and an error notifying enhancement

barryf opened the pull request:
1. Add repo (template) to Heroku deploy button link.
2. Take out `bundle` command in app package which was causing an error.
3. Specify a channel/user to notify in case things go wrong.

(3) was added when initial deployment crashed: it was trying to archive a channel with a single-channel user and borking. Getting a DM means I realise it's broken rather than silently failing each hour.

barryf committed:
Specify this repo on the Heroku deploy link.

barryf committed:
Removing reference to bundle post-deploy.

barryf committed:
Require ARCHIVEBOT_NOTIFY for a user to DM any errors

barryf committed:
Updated README and app.json to reflect new notify env var.

barryf committed:
Move onErrorNotify definition into function.

timblair commented on README.md:
```diff
@@ -23,7 +23,7 @@ godep go build
 using the Heroku Scheduler add-on.
 
-[![Deploy](https://www.herokucdn.com/deploy/button.png)](https://heroku.com/deploy)
+[![Deploy to Heroku](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy?template=https://github.com/timblair/slack-archivebot)
```
Please could you move the URLs to reference links to avoid the long line.

timblair commented on README.md:
```diff
@@ -39,6 +39,7 @@ The following environment variables are used to configure the script:
   for your team.
 * `ARCHIVEBOT_INACTIVITY_DAYS`: the number of days' inactivity after which to
   archive a channel (default: `30`).
+* `ARCHIVEBOT_NOTIFY`: a Slack user or channel (e.g. `#general` or `@tblair`) to notify when something goes wrong.
```
Please wrap this line at 78 characters (or after the closing bracket, whatever that point it).

timblair commented on bot.go:
```diff
@@ -59,7 +59,12 @@ func archiveChannels(api *slack.Slack, c []slack.Channel, reason string) {
 			defer wg.Done()
 			if err := api.ArchiveChannel(c.Id); err != nil {
-				log.Printf("Error archiving #%s (%s): %s\n", c.Name, c.Id, err)
+				message := fmt.Sprintf("Error archiving channel #%s (%s): %s\n", c.Name, c.Id, err)
```
I appreciate that we're quite nested here, but is there a nice way to split this line to keep within the line length?

timblair commented on bot.go:
```diff
@@ -59,7 +59,12 @@ func archiveChannels(api *slack.Slack, c []slack.Channel, reason string) {
+				// send error message in a DM to onErrorNotify user/channel
+				onErrorNotify := os.Getenv("ARCHIVEBOT_NOTIFY")
+				params := slack.PostMessageParameters{}
+				api.PostMessage(onErrorNotify, message, params)
```
I'm guessing that this API call will fail if the `ARCHIVEBOT_NOTIFY` environment variable is empty, but that the error will be ignored?  I'd rather be explicit about only attempting the notification if there's a user/channel defined, and it should also log an error if the call fails.

timblair reviewed (changes requested):
I like the ideas.  Just a few small bits to go back to.

barryf committed:
Break line at 80 chars in README.

barryf committed:
Broke line to wrap under 80 chars.

There doesn't seem to be a fixed line length but following
your request --
https://github.com/golang/go/wiki/CodeReviewComments#line-length

barryf committed:
Allow for ARCHIVEBOT_NOTIFY to not be set.

barryf committed:
Moved Heroku link to reference style

(I didn't change the image URL because my Markdown
parsers didn't recognise it. YMMV).

barryf commented:
I've pushed new commits that address your requests. Can you review?

timblair commented on bot.go:
```diff
@@ -59,7 +59,15 @@ func archiveChannels(api *slack.Slack, c []slack.Channel, reason string) {
+				onErrorNotify := os.Getenv("ARCHIVEBOT_NOTIFY")
+				if onErrorNotify != "" {
+					params := slack.PostMessageParameters{}
+					api.PostMessage(onErrorNotify, message, params)
```
Please log a message if this call fails.

timblair reviewed (changes requested):
One last thing, then it looks good to merge to me.

barryf committed:
Log error if Slack message post fails.

@tblair Does this cover logging a message? (this is stretching my Go knowledge somewhat)
